### PR TITLE
python3Packages.pyhaversion: 3.4.2 -> 20.12.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 20.12.1.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash all the ``python3Packages.pyhaversion`` commits.

**Comment by fabaff** [comment]
> Please squash all the python3Packages.pyhaversion commits.

Done. Squashed back till update but leave update commit in place in case we have to revert back to 3.4.2.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108777` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>python37Packages.pyhaversion: log <a href="https://termbin.com/mzt3">https://termbin.com/mzt3</a></li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.awesomeversion</li>
<li>python38Packages.awesomeversion</li>
<li>python39Packages.awesomeversion</li>
</ul>
</details>

File: pkgs/development/python-modules/awesomeversion/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, requests
+}:
+
+buildPythonPackage rec {
+  pname = "awesomeversion";
+  version = "20.12.5";
```
Comment (jonringer) [inline pkgs/development/python-modules/awesomeversion/default.nix:10]:
```suggestion
  version = "20.12.5";
  disabled = pythonOlder "3.8";
```

File: pkgs/development/python-modules/awesomeversion/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, requests
```
Comment (jonringer) [inline pkgs/development/python-modules/awesomeversion/default.nix:5]:
```suggestion
, requests
, pythonOlder
```

**Review by jonringer — COMMENTED** [review]
```
https://github.com/NixOS/nixpkgs/pull/108777

1 package failed to build:
python37Packages.pyhaversion

3 packages built:
python37Packages.awesomeversion python38Packages.awesomeversion python39Packages.awesomeversion
```

File: pkgs/development/python-modules/awesomeversion/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/awesomeversion/default.nix:10]:
```
  Requirement already satisfied: aiohttp<4.0,>=3.6.1 in /nix/store/pyyzpgwph5q1aajs9f9ni79ddbb096wr-python3.7-aiohttp-3.7.3/lib/python3.7/site-packages (from pyhaversion==master) (3.7.3)
  ERROR: Package 'pyhaversion' requires a different Python: 3.7.9 not in '>=3.8.0'
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>python37Packages.pyhaversion: log <a href="https://termbin.com/d9fr">https://termbin.com/d9fr</a></li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.awesomeversion</li>
<li>python38Packages.awesomeversion</li>
<li>python39Packages.awesomeversion</li>
</ul>
</details>

File: pkgs/development/python-modules/awesomeversion/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/awesomeversion/default.nix:11]: Added. Thanks for the input.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 108777` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.awesomeversion</li>
    <li>python39Packages.awesomeversion</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@fabaff please fix the merge conflict.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108777` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.awesomeversion</li>
<li>python39Packages.awesomeversion</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.awesomeversion</li>
<li>python39Packages.awesomeversion</li>
</ul>
</details>

**Pull request merged**